sssd_test_framework/roles/generic.py

```diff
@@ -1275,3 +1311,61 @@ def policy(self, logon_rights: dict[str, list[Any]], cfg: dict[str, Any] | None
         :rtype: GenericGPO
         """
         pass
+
+
+class GenericPasswordPolicy(ABC, BaseObject):
+    """
+    Password policy management.
+    """
+
+    @abstractmethod
+    def complexity(self, enable: bool) -> GenericPasswordPolicy:
+        """
+        Enable or disable password complexity.
+
+        :param enable: Enable or disable password complexity.
+        :type enable: bool
+        :return ADPasswordPolicy object.
+        :rtype: ADPasswordPolicy
+        """
+        pass
+
+    @abstractmethod
+    def lockout(self, duration: int, attempts: int) -> GenericPasswordPolicy:
+        """
+        Set lockout duration and login attempts.
+
+        :param duration: Duration of lockout in seconds.
+        :type duration: int
+        :param attempts: Number of login attempts.
+        :type attempts: int
+        :return: ADPasswordPolicy object.
+        :rtype: ADPasswordPolicy
+        """
+        pass
+
+    @abstractmethod
+    def age(self, minimum: int, maximum: int) -> GenericPasswordPolicy:
+        """
+        Set maximum and minimum password age.
+
+        :param minimum: Minimum password age in seconds.
+        :type minimum: int
+        :param maximum: Maximum password age in seconds.
+        :type maximum: int
+        :return: ADPasswordPolicy object.
+        :rtype: ADPasswordPolicy
+        """
+        pass
+
+    @abstractmethod
+    def requirements(self, length: int) -> GenericPasswordPolicy:
```

Comment:
It is to add parameters in the future, but we can't use *args, or **kwargs because all the parameters are different per provider.  I'm not even sure we will need the other ones, so I'll change this to length and then we can change later if we need to.

Reply:
It's not being used right now, maybe we just leave as it.